phpstan.el

```diff
@@ -28,19 +28,83 @@
 ;; https://github.com/phpstan/phpstan
 
 ;;; Code:
+(require 'php-project)
 (require 'flycheck nil)
 
+
+;; Variables:
+
+;;;###autoload
+(progn
+  (defvar phpstan-config-file nil)
+  (make-variable-buffer-local 'phpstan-config-file)
+  (put 'phpstan-config-file 'safe-local-variable
+       #'(lambda (v) (if (consp v)
+                         (and (eq 'root (car v)) (stringp (cdr v)))
+                       (null v) (stringp v)))))
+
+;;;###autoload
+(progn
+  (defvar phpstan-level "0")
+  (make-variable-buffer-local 'phpstan-level)
+  (put 'phpstan-level 'safe-local-variable
+       #'(lambda (v) (or (null v)
+                         (integerp v)
+                         (and (stringp v)
+                              (string-match-p "\\`[1-9][0-9]*\\'" v))))))
+
+;;;###autoload
+(progn
+  (defvar phpstan-executable nil)
+  (make-variable-buffer-local 'phpstan-executable)
+  (put 'phpstan-executable 'safe-local-variable
+       #'(lambda (v) (if (consp v)
+                         (and (eq 'root (car v)) (stringp (cdr v)))
+                       (null v) (stringp v)))))
+
+;; Functions:
+(defun phpstan-get-config-file ()
+  "Return path to phpstan configure file or `NIL'."
+  (if phpstan-config-file
+      (if (and (consp phpstan-config-file)
+               (eq 'root (car phpstan-config-file)))
```

Comment:
It is not clear what the structure of `phpstan-config-file` should be.  I think user options should at least be defined as defcustoms with a type annotation.